public/app/plugins/panel/xychart/XYChartPanel2.tsx

```diff
@@ -13,116 +22,225 @@ import {
   VizTooltipContainer,
 } from '@grafana/ui';
 import { FacetedData } from '@grafana/ui/src/components/uPlot/types';
+import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
 
 import { TooltipView } from './TooltipView';
 import { XYChartOptions } from './models.gen';
-import { prepData, prepScatter } from './scatter';
+import { prepData, prepScatter, ScatterPanelInfo } from './scatter';
 import { ScatterHoverEvent, ScatterSeries } from './types';
 
 type Props = PanelProps<XYChartOptions>;
-type State = {
-  error?: string;
-  series: ScatterSeries[];
-  builder?: UPlotConfigBuilder;
-  facets?: FacetedData;
-  hover?: ScatterHoverEvent;
-};
+const TOOLTIP_OFFSET = 10;
 
-export class XYChartPanel2 extends PureComponent<Props, State> {
-  state: State = {
-    series: [],
-  };
+export const XYChartPanel2: React.FC<Props> = (props: Props) => {
+  const [error, setError] = useState<string | undefined>();
+  const [series, setSeries] = useState<ScatterSeries[]>([]);
+  const [builder, setBuilder] = useState<UPlotConfigBuilder | undefined>();
+  const [facets, setFacets] = useState<FacetedData | undefined>();
+  const [hover, setHover] = useState<ScatterHoverEvent | undefined>();
+  const [shouldDisplayCloseButton, setShouldDisplayCloseButton] = useState<boolean>(false);
 
-  componentDidMount() {
-    this.initSeries(); // also data
-  }
+  const isToolTipOpen = useRef<boolean>(false);
```

Comment:
This needs to be a ref?

Reply:
I tried with state but it didn't work out. Not entirely sure why but I'm going to leave it a ref for now and look over it later.